packages/flutter/lib/src/widgets/linked_text.dart

```diff
@@ -0,0 +1,759 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'text.dart';
+
+/// Signature for a callback that passes a [String] representing a link that has
+/// been tapped.
+typedef LinkTapCallback = void Function(String linkString);
```

Comment:
Maybe easier to use ValueChanged

Reply:
I left some questions about this approach on your other comment: https://github.com/flutter/flutter/pull/125927#discussion_r1312226825.